test/Test/Database/LSMTree/Internal/Run.hs

```diff
@@ -209,3 +229,98 @@ prop_WriteAndOpen fs hbio wb =
 
       -- TODO: return a proper Property instead of using assertEqual etc.
       return (property True)
+
+-- | Writing and loading a 'WriteBuffer' gives the same in-memory
+--   representation as the original run.
+prop_WriteAndOpenWriteBuffer ::
+     FS.HasFS IO h
+  -> FS.HasBlockIO IO h
+  -> RunData KeyForIndexCompact SerialisedValue SerialisedBlob
+  -> IO Property
+prop_WriteAndOpenWriteBuffer hfs hbio rd = do
+  withTempRegistry $ \reg -> do
+    -- Serialise run data as write buffer:
+    let srd = serialiseRunData rd
+    let inPaths = WrapRunFsPaths $ simplePath 1111
+    let resolve (SerialisedValue x) (SerialisedValue y) = SerialisedValue (x <> y)
+    withRunDataAsWriteBuffer hfs resolve inPaths srd $ \wb wbb -> do
+      -- Write write buffer to disk:
+      let wbPaths = WrapRunFsPaths $ simplePath 1312
+      withSerialisedWriteBuffer hfs hbio wbPaths wb wbb $ do
+        -- Read write buffer from disk:
+        (wb', wbb') <- readWriteBuffer reg resolve hfs hbio wbPaths
+        assertEqual "k/ops" wb wb'
+        releaseRef wbb'
```

Comment:
This is not comparing the blobs, only the key/ops. Something similar to `readKOps` as used in `prop_WriteAndOpen` should work, I think?